### ISSUE TITLE
Support for non-convolutional batch normalization

### ISSUE THREAD

**Issue by rinuboney** [google/prettytensor#2]
any plans to implement it soon?

**Comment by eiderman** [comment]
No plans to implement it since the underlying tf op doesn't support it.  As a workaround, you can reshape (below uses the shorthand for reshaping):

    y = x.fully_connected(100).reshape('__11').batch_normalize(...).reshape('__')

Or if you want to use this pattern a lot, you can register a custom method from within your code:


    # assign_defaults allows these to be set using a pt.defaults_scope across an entire model.
    @pt.Register(assign_defaults=('l2loss', 'activation_fn', 
         'learned_moments_update_rate', 'variance_epsilon', 'scale_after_normalization', 'phase'))
    def full_with_normalization(layer_size, 
                                               l2loss=None, 
                                               activation_fn=None, 
                                               learned_moments_update_rate=None, 
                                               variance_epsilon=None, 
                                               scale_after_normalization=None):
     return (x.fully_connected(layer_size, l2loss=l2loss, activation_fn=activation_fn)
                 .reshape('__11')
                 .batch_normalize(learned_moments_update_rate=learned_moments_update_rate,
                                             variance_epsilon=variance_epsilon,
                                             scale_after_normalization=scale_after_normalization)
                 .reshape('__')

**Comment by eiderman** [comment]
Added in 0.6.2
